internal/sinks/alertmanager.go

```diff
@@ -0,0 +1,160 @@
+package sinks
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/spectrocloud-labs/validator/api/v1alpha1"
+)
+
+type AlertmanagerSink struct {
+	client Client
+	log    logr.Logger
+
+	endpoint string
+	username string
+	password string
+}
+
+type Alert struct {
+	Annotations map[string]string `json:"annotations"`
+	Labels      map[string]string `json:"labels"`
+}
+
+var (
+	InvalidEndpoint  = errors.New("invalid Alertmanager config: endpoint scheme and host are required")
+	EndpointRequired = errors.New("invalid Alertmanager config: endpoint required")
+)
+
+func (s *AlertmanagerSink) Configure(c Client, config map[string][]byte) error {
+	// endpoint
+	endpoint, ok := config["endpoint"]
+	if !ok {
+		return EndpointRequired
+	}
+	u, err := url.Parse(string(endpoint))
+	if err != nil {
+		return errors.Wrap(err, "invalid Alertmanager config: failed to parse endpoint")
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return InvalidEndpoint
+	}
+	if u.Path != "" {
+		u.Path = ""
```

Comment:
do you think it would be worth it to log a warning that the path is being overridden here? i can go either way with this, but think it might be worth it so that whoever sets up the sink realizes this is happening when they pass in a "bad" endpoint.